src/main/java/ch/njol/skript/expressions/ExprName.java

```diff
@@ -107,9 +91,14 @@ public class ExprName extends SimplePropertyExpression<Object, String> {
 		if (Skript.classExists("net.kyori.adventure.text.Component") &&
 			Skript.methodExists(Bukkit.class, "createInventory", InventoryHolder.class, int.class, Component.class))
 			serializer = BungeeComponentSerializer.get();
-		register(ExprName.class, String.class, "(1:name[s])", "offlineplayers/entities/inventories/nameds");
-		register(ExprName.class, String.class, "(2:(display|nick|chat|custom)[ ]name[s])", "offlineplayers/entities/inventories/nameds");
-		register(ExprName.class, String.class, "(3:(player|tab)[ ]list name[s])", "players");
+
+		Skript.registerExpression(ExprName.class, String.class, ExpressionType.COMBINED,
+			"[the] name[s] of %offlineplayers/entities/inventories/nameds%",
+			"%offlineplayers/entities/inventories/nameds%'[s] name[s]",
+			"[the] (display|nick|chat|custom)[ ]name[s] of %offlineplayers/entities/inventories/nameds%",
+			"%offlineplayers/entities/inventories/nameds%'[s] (display|nick|chat|custom)[ ]name[s]",
+			"[the] (player|tab)[ ]list name[s] of %players%",
+			"%players%'[s] (player|tab)[ ]list name[s]");
```

Comment:
You should be able to use PropertyExpression#getPatterns() for each of the 3

Reply:
son of a beep